crates/turborepo-lib/src/diagnostics.rs

```diff
@@ -0,0 +1,523 @@
+            options,
+            tx,
+        ))
+        .await
+        .ok()?; // if the channel is closed, we can't request
+
+        Some(rx)
+    }
+
+    /// suspends the terminal output and returns a pair of channels
+    ///
+    /// the first one signifies to the diagnostic when it is safe to render
+    /// and the second one is used by the diagnostic to notify the renderer
+    /// when it is done
+    async fn suspend(chan: &MpscSender<DiagnosticMessage>) -> Option<(Receiver<()>, Sender<()>)> {
+        let (stopped_tx, stopped_rx) = oneshot::channel();
+        let (resume_tx, resume_rx) = oneshot::channel();
+
+        chan.send(DiagnosticMessage::Suspend(
+            Self::NAME.to_string(),
+            stopped_tx,
+            resume_rx,
+        ))
+        .await
+        .ok()?; // if the channel is closed, we can't suspend
+
+        Some((stopped_rx, resume_tx))
+    }
+
+    // types of exit
+
+    /// the diagnostic is done
+    async fn done(chan: MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::Done(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+
+    /// the diagnostic failed
+    async fn failed(chan: MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::Failed(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+
+    /// the diagnostic is not applicable
+    async fn not_applicable(chan: MpscSender<DiagnosticMessage>, message: String) {
+        chan.send(DiagnosticMessage::NotApplicable(
+            Self::NAME.to_string(),
+            message.to_string(),
+        ))
+        .await
+        .ok(); // channel closed, ignore
+    }
+}
+
+pub struct GitDaemonDiagnostic;
+
+impl Diagnostic for GitDaemonDiagnostic {
+    const NAME: &'static str = "git.daemon";
+
+    fn execute(&self, chan: MpscSender<DiagnosticMessage>) {
+        tokio::task::spawn(async move {
+            if !(cfg!(target_os = "windows") || cfg!(target_os = "macos")) {
+                // the git daemon is not implemented on unix
+                Self::not_applicable(chan, "Git FS Monitor (not available on unix)".to_string())
+                    .await;
+                return;
+            }
+
+            Self::started(&chan, "Git FS Monitor".to_string()).await;
+
+            let futures: Result<Vec<Vec<u8>>, _> = join_all(
+                [
+                    &["--version"][..],
+                    &["config", "--get", "core.fsmonitor"][..],
+                    &["config", "--get", "core.untrackedcache"][..],
+                ]
+                .into_iter()
+                .map(|args| async move {
+                    // get the current setting
+                    let stdout = Stdio::piped();
+
+                    let command = Command::new("git")
```

Comment:
We might want to consider using `which` here. That generally avoids some bugs with finding the correct command